src/brpc/progressive_reader.h

```diff
@@ -84,6 +85,8 @@ class ReadableProgressiveAttachment : public SharedObject {
     // Any error occurred should destroy the reader by calling r->Destroy().
     // r->Destroy() should be guaranteed to be called once and only once.
     virtual void ReadProgressiveAttachmentBy(ProgressiveReader* r) = 0;
+    virtual SocketId GetSocketId() = 0;
+    virtual void SetSocketId(SocketId id) = 0;
```

Comment:
The  ReadableProgressiveAttachment interface doesn't need a virtual SetSocketId function.
The SetSocketId function can be only a member of HttpContext.